world/roll.py

```diff
@@ -60,20 +62,31 @@ def __init__(self, caller=None, stat=None, skill=None, difficulty=15, stat_list=
             stat_list = [ob.lower() for ob in stat_list]
             # look up each stat from supplied caller, adds to stats dict
             for somestat in stat_list:
-                self.stats[somestat] += self.character.attributes.get(somestat, 0)
+                if self.retainer is None:
+                    self.stats[somestat] += self.character.attributes.get(somestat, 0)
+                else:
+                    self.stats[somestat] += self.retainer.dbobj.attributes.get(somestat, 0)
```

Comment:
accessing stats via `attributes` is also deprecated, please get via traitshandler